state/unit.go

```diff
@@ -359,23 +362,32 @@ func removeUnitStateOp(mb modelBackend, globalKey string) txn.Op {
 }
 
 // SetState replaces the currently stored state for a unit with the contents
-// of the provided unitState parameter.
-func (u *Unit) SetState(unitState map[string]string) error {
-	modelOp := u.SetStateOperation(unitState)
+// of the provided unitState and uniterState parameters.
+//
+// TODO (hml) 03-18-2020
+// Change the arg to a struct as storage and relation state is added.
+func (u *Unit) SetState(unitState map[string]string, uniterState string) error {
+	modelOp := u.SetStateOperation(unitState, uniterState)
 	return u.st.ApplyOperation(modelOp)
 }
 
 // SetStateOperation returns a ModelOperation for replacing the currently
 // stored state for a unit with the contents of the provided unitState
-// parameter.
-func (u *Unit) SetStateOperation(unitState map[string]string) ModelOperation {
-	return &unitSetStateOperation{u: u, newState: unitState}
+// and uniterState parameters.
+//
+// TODO (hml) 03-18-2020
+// Change the arg to a struct as storage and relation state is added.
+func (u *Unit) SetStateOperation(unitState map[string]string, uniterState string) ModelOperation {
+	return &unitSetStateOperation{u: u, newState: unitState, newUniterState: uniterState}
 }
 
 // State returns the persisted state for a unit.
-func (u *Unit) State() (map[string]string, error) {
+//
+// TODO (hml) 03-18-2020
+// Return a structure instead as storage and relation state is added.
```

Comment:
How about extracting all the Get/SetState bits into a standalone PR so this TODO can be addressed? I would prefer we minimize the amount of TODOS given that 2.8 is near. Doing so would make the changes in this PR easier to review.

Reply:
Keep in mind that we also still need to support storing the uniter relation state data so I think having a wrapper struct here that also includes an extra `map[int]interface{}`or `map[int]map[string]interface{}` field for this would be beneficial before we proceed with the actual uniter changes.

Reply:
I'll pull the state piece out into a separate PR.  I was planning to leave all of the uniter state pieces, uniter, relation, storage etc, as a string of the marshaled yaml of the struct if possible.  As done for the uniter internal state.

Reply:
SGTM; however I don't think it will be as easy for the relation/storage bits as it is a yaml per relation (not sure what's the equivalent for storage) so maybe a `map[int]string` (int = relation id) then for the individual serialized payloads?